corehunter-base/src/main/java/org/corehunter/data/simple/SimpleGenotypeVariantData.java

```diff
@@ -399,11 +401,27 @@ public final static void writeData(Path filePath,
         
         switch (format) {
             case DIPLOID:
-                writeDiploidData(filePath, genotypicData, type);
-                break ;
+                if (genotypicData instanceof GenotypeVariantData) {
```

Comment:
Is there any reason to make the `writeData` method static? I think it would be more clean to make those regular instance methods so that you can override for biallelic data and avoid using `instanceof`.